include/nudb/impl/basic_store.ipp

```diff
@@ -217,7 +216,7 @@ insert(
     if(ecb_)
     {
         ec = ec_;
-        return false;
+        return;
     }
     // Data Record
     if(size == 0)
```

Comment:
`logic_error` and `domain_error` aren't meant to be caught, they are meant to terminate the app since they are completely avoidable by the programmer. Making them errors would mislead users into thinking its okay to call those functions in a way that violates the preconditions. The alternative is to make calls to `fetch` without an open database undefined behavior. I think the exception is better.